src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1474,8 +1599,14 @@ private void ScanTagInDocumentContext()
                     // Checking to see if we meet the conditions of a special '!' tag: <!DOCTYPE, <![CDATA[, <!--.
                     if (!IsBangEscape(lookahead: 1))
                     {
+                        if (Lookahead(2)?.Type == HtmlSymbolType.DoubleHyphen)
```

Comment:
Why this?

Reply:
We want to make sure that the '<' will not be part of whatever content was before it, in case of a comment